commands/operator-sdk/cmd/scorecard/basic_tests.go

```diff
@@ -0,0 +1,186 @@
+		err := runtimeClient.Get(context.TODO(), types.NamespacedName{Namespace: SCConf.Namespace, Name: name}, &obj)
+		if err != nil {
+			return false, fmt.Errorf("error getting custom resource: %v", err)
+		}
+		pass := true
+		if obj.Object["spec"] == nil {
+			pass = false
+			specPoints = 0
+		} else {
+			specPoints = 1
+		}
+		if obj.Object["status"] == nil {
+			pass = false
+			statusPoints = 0
+		} else {
+			statusPoints = 1
+		}
+		return pass, nil
+	})
+	if !noStore {
+		testSpec.earnedPoints = specPoints
+		testStat.earnedPoints = statusPoints
+		scTests = append(scTests, testSpec)
+		scTests = append(scTests, testStat)
+	}
+	if err != nil && !reflect.DeepEqual(err, wait.ErrWaitTimeout) {
+		return err
+	}
+	return nil
+}
+
+// TODO: user specified tests for operators
+
+// checkStatusUpdate looks at all fields in the spec section of a custom resource and attempts to modify them and
+// see if the status changes as a result. This is a bit prone to breakage as this is a black box test and we don't
+// know much about how the operators we are testing actually work and may pass an invalid value. In the future, we
+// should use user-specified tests
+func checkStatusUpdate(runtimeClient client.Client, obj unstructured.Unstructured) error {
+	test := scorecardTest{testType: basicOperator, name: "Operator actions are reflected in status", maximumPoints: 1}
+	err := runtimeClient.Get(context.TODO(), types.NamespacedName{Namespace: SCConf.Namespace, Name: name}, &obj)
+	if err != nil {
+		return fmt.Errorf("error getting custom resource: %v", err)
+	}
+	if obj.Object["status"] == nil || obj.Object["spec"] == nil {
+		scTests = append(scTests, test)
+		return nil
+	}
+	statCopy := make(map[string]interface{})
+	for k, v := range obj.Object["status"].(map[string]interface{}) {
+		statCopy[k] = v
+	}
+	pass := true
+	specMap := obj.Object["spec"].(map[string]interface{})
+	for k, v := range specMap {
+		switch t := v.(type) {
+		case int64:
+			specMap[k] = specMap[k].(int64) + 1
+		case float64:
+			specMap[k] = specMap[k].(float64) + 1
+		case string:
+			// TODO: try and find out how to make this better
+			// Since strings may be very operator specific, this test may not work correctly in many cases
+			specMap[k] = fmt.Sprintf("operator sdk test value %f", rand.Float64())
+		case bool:
+			specMap[k] = !specMap[k].(bool)
+		case []interface{}: // TODO: Decide how this should be handled
+		default:
+			fmt.Printf("Unknown type for key (%s) in status: (%v)\n", k, reflect.TypeOf(t))
+		}
+		obj.Object["spec"] = specMap
+		runtimeClient.Update(context.TODO(), &obj)
+		err := wait.Poll(time.Second*1, time.Second*15, func() (done bool, err error) {
+			runtimeClient.Get(context.TODO(), types.NamespacedName{Namespace: SCConf.Namespace, Name: name}, &obj)
+			if err != nil {
+				return false, err
+			}
+			return !reflect.DeepEqual(statCopy, obj.Object["status"]), nil
+		})
+		if err != nil {
+			pass = false
```

Comment:
you only set pass once and then use that bool to set the points. I think that it could all be in one if statement